libs/community/langchain_community/graph_vectorstores/extractors/hierarchy_link_extractor.py

```diff
@@ -0,0 +1,74 @@
+from typing import Callable, List, Set
+
+from langchain_core.documents import Document
+from langchain_core.graph_vectorstores.links import Link
+
+from langchain_community.graph_vectorstores.extractors.link_extractor import (
+    LinkExtractor,
+)
+from langchain_community.graph_vectorstores.extractors.link_extractor_adapter import (
+    LinkExtractorAdapter,
+)
+
+# TypeAlias is not available in Python 2.9, we can't use that or the newer `type`.
+HierarchyInput = List[str]
+
+
+class HierarchyLinkExtractor(LinkExtractor[HierarchyInput]):
+    def __init__(
+        self,
+        kind: str = "hierarchy",
```

Comment:
I generally suggest using `*` for any named arguments w/ defaults to make implementations future proof against refactors that may add new named arguments

```suggestion
        *,
        kind: str = "hierarchy",
```